src/toil_scripts/adam_uberscript/adam_uberscript.py

```diff
@@ -465,10 +475,30 @@ def collect_realtime_metrics(params, threshold=0.5):
                     try:
                         with cluster_size_lock:
                             cluster_size = get_cluster_size(params.cluster_name)
-                            if cluster_size > get_desired_cluster_size(domain):
-                                log.info('Terminating Instance: {}'.format(instance_id))
-                                log.info('Killing instance {0}\n'.format(instance_id))
-                                conn.terminate_instances(instance_ids=[instance_id])
+                            desired_cluster_size = get_desired_cluster_size(domain)
+                            if cluster_size > desired_cluster_size:
+                                log.info('Cluster size (%d) is larger than requested (%d).'
+                                         'Terminating idle instance %s.',
+                                         cluster_size,
+                                         desired_cluster_size,
+                                         instance_id)
+                                
+                                cmd = ['cgcloud',
+                                       'terminate',
+                                       '--instance-id', instance_id,
+                                       '--cluster-name', params.cluster_name,
+                                       '--quick',
+                                       'toil']
+
+                                try:
+                                    check_call(cmd)
+                                    log.info("Successfully terminated instance via %s.",
+                                             " ".join(cmd))
```

Comment:
I usually do `log.info("Successfully terminated instance via %r.", cmd)` because that's not lossy in case there are spaces in any of the elements of cmd.